test.sh

```diff
@@ -74,20 +74,19 @@ function die() {
 
 function run_unit_tests() {
   if [ "${TRAVIS}" == "true" ]; then
-
-    # The deps variable is the imports of the packages under test that
-    # are not stdlib packages. We can then install them with the race
-    # detector enabled to prevent our individual `go test` calls from
-    # building them multiple times.
-    all_shared_imports=$(go list -f '{{ join .Imports "\n" }}' ${TESTPATHS} | sort | uniq)
-    deps=$(go list -f '{{ if not .Standard }}{{ .ImportPath }}{{ end }}' ${all_shared_imports})
-    echo "go installing race detector enabled dependencies"
-    go install -race $deps
-
-    # Run each test by itself for Travis, so we can get coverage
+    # Run the full suite of tests once with the -race flag. Since this isn't
```

Comment:
Can you make this a separate test phase? E.g. RUN=cover ? Then we can shift it around as necessary to balance out the work, or put it in a separate Travis task.